gum/arch-arm/gumarmrelocator.c

```diff
@@ -345,3 +385,28 @@ gum_arm_relocator_rewrite_branch_imm (GumArmRelocator * self,
 
   return TRUE;
 }
+
+static gboolean
+gum_arm_relocator_rewrite_pc_relative_ldr (GumArmRelocator * self,
+                                           GumCodeGenCtx * ctx)
+{
+  cs_insn * ci = ctx->capstone_insn;
+  gint dest_reg, base_reg, value;
+
+  dest_reg = gum_capstone_reg_to_arm_reg (ci->detail->arm.operands[0].reg);
+  base_reg = ci->detail->arm.operands[1].mem.base;
+  value = *((gint *) GSIZE_TO_POINTER (ctx->insn->pc + ci->detail->arm.operands[1].mem.disp));
```

Comment:
Hmm, is this safe in case of self-modifying code? Shouldn't we just load the target register with what the PC would have been at the original instruction, then add to it, and dereference?